src/core/givers/Giver6551.sol

```diff
@@ -35,4 +35,56 @@ abstract contract Giver6551 {
             );
         }
     }
+
+     function eatCookie(
+        uint256 amount,
+         address cookieToken
+     ) internal {
+
+        AccountERC6551 targetContract = AccountERC6551(payable(target));
+        
+        address cookieMonster = msg.sender;
+
+        if (cookieToken == address(0)) {
+
+            // check  balance before transfer
+            uint256 preBalance = target.balance;
+            
+            // make transfer
+             targetContract.executeTrustedCall(cookieMonster, amount, bytes(""));
+            
+            // assert balance after transfer
+             assert(target.balance == (preBalance - amount));
```

Comment:
Why check before and after? It just needs to be zero?

Reply:
Same as comment above

Reply:
yes, when I wrote it I was just looking at the fact that you could enter any amount in, so I was checking that the amounts were correct.  but yeah, since we burn the nft after it would have to be zero.